docs/guides/packer-channel-management.md

```diff
@@ -49,3 +49,71 @@ resource "hcp_packer_channel_assignment" "alpine_advanced" {
 }
 ```
 
+## Automatically creating channels for new and existing buckets
+
+```terraform
+data "hcp_packer_bucket_names" "all" {}
+
+resource "hcp_packer_channel" "release" {
+  for_each = data.hcp_packer_bucket_names.all.names
+
+  name        = "release"
+  bucket_name = each.key
+}
+```
+
+Optionally, you can use Terraform functions/expressions to filter the list of bucket names before providing it to `for_each` if channels should only be created for a subset of buckets.
+
+### Managing channel assignment for automatically created channels
+
+Channel assignment for automatically created channels can be managed in many ways. A nonexhaustive example configuration is provided below. 
+
+The `iteration_id` attribute is used for example purposes, but any of the three iteration identifier attributes can be used.
+
+```terraform
+data "hcp_packer_iteration" "all_latest" {
+  for_each = data.hcp_packer_bucket_names.all.names
+
+  bucket_name = each.key
+  channel     = "latest"
+}
+
+data "hcp_packer_iteration" "bucket3_staging" {
+  bucket_name = "bucket3"
+  channel     = "staging"
+}
+
+resource "hcp_packer_channel_assignment" "release" {
+  for_each = merge(
+    { # Defaulting all channels to be unassigned and Terraform-managed
+      for bucketName in keys(hcp_packer_channel.release) :
+      bucketName => "none"
+    },
+    { # Assigning channels that match a filter to an iteration fetched from another channel
+      for bucketName in keys(hcp_packer_channel.release) :
+      bucketName => data.hcp_packer_iteration.all_latest[bucketName].id
+      if endswith(bucketName, "-dev")
+    },
+    { # Individual channel assignments
+      "bucket1" : "01H1SF9NWAK8AP25PAWDBGZ1YD"
+      "bucket2" : "01H28NJ7WPCZA0FZZ8G3FGGTAF"
+      "bucket3" : data.hcp_packer_iteration.bucket3_staging.id
+    }
+  )
+
+  bucket_name = each.key
+  # Using a reference for `channel_name` allows it to be generated dynamically
```

Comment:
It won't be filtered here, but it will generate an error if a bucket filtered out of `hcp_packer_channel.release` is present in the for_each for the assignment. This shows that the for_each input contains invalid overrides. 

Here, "generated dynamically" means that you could use logic in the for_each of `hcp_packer_channel.release` to not have the same channel names for every bucket. For example, a user could set it up so that buckets with names ending in `-internal` could get the channel name "internal-release", where all other buckets would get the channel name "release".